pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -191,12 +221,13 @@ func (m *rangefeedMuxer) startSingleRangeFeed(
 	}()
 
 	// Start a retry loop for sending the batch to the range.
-	for r := retry.StartWithCtx(ctx, m.ds.rpcRetryOptions); r.Next(); {
-		maybeReleaseTransport()
-
+	for stream.retry.Next() {
 		// If we've cleared the descriptor on failure, re-lookup.
-		if !token.Valid() {
-			var err error
+		if !stream.token.Valid() {
```

Comment:
Most of this loop checks and modifies `stream.Something` fields. This suggests that much of this code logically belongs to the `stream` type, rather than muxer. Could this be encapsulated in the `stream` as a few stream lifecycle methods, and the muxer would just orchestrate?

Reply:
Perhaps it can; it's more of a result of inheriting "dna" from regular rangefeed;
Do you feel strongly that such refactoring should be done right now?